src/minitwit.go

```diff
@@ -293,8 +319,51 @@ func handleLogin(w gin.ResponseWriter, r *http.Request, c *gin.Context) {
 		}
 
 	}
+	cookie, err := getCookie(c)
+	if err != nil {
+		out, err := loginTemplate.Execute(gonja.Context{"g": ""})
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+		w.Write([]byte(out))
+	} else {
+		out, err := loginTemplate.Execute(gonja.Context{"g": cookie})
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+		w.Write([]byte(out))
+	}
+}
 
-	out, err := loginTemplate.Execute(gonja.Context{"g": ""})
+func handleRegister(w gin.ResponseWriter, r *http.Request, c *gin.Context) {
```

Comment:
Needs to return "/" for s user who is already logged in, according to line 98 in minitwit.py
```
  print("We got a visitor from: " + str(request.remote_addr))
  if not g.user:
      return redirect(url_for('public_timeline'))
```